File: Task 3/Java/logistics/src/main/java/com/logistics/Components/Product/Packaging.java
package com.logistics.Components.Product;

public class Packaging {

    private int units;
    private float costPerUnit;

    public Packaging() {
        units = 0;
        costPerUnit = 0;
    }

    public Packaging(int units, float costPerUnit) {
        this.units = units;
        this.costPerUnit = costPerUnit;
    }

    public Packaging(Packaging other) {
        this.units = other.units;
        this.costPerUnit = other.costPerUnit;
    }

    // Getters
    public int getUnits() { return units; }
    public float getCostPerUnit() { return costPerUnit; }

    // Setters
    public void setUnits(int units) { this.units = units; }
    public void setCostPerUnit(float costPerUnit) { this.costPerUnit = costPerUnit; }

    public float calculateTotalCost() {
        return units * costPerUnit;
    }

    public String display_product_info() {
        return "Packaging Units: " + units + 
                "\nCost Per Unit: " + costPerUnit;
    }
}

File: Task 1/C++/Components/Product/Packaging.cpp
#include "Packaging.h"

Packaging::Packaging(){
    packaging_units = 0;
    packaging_cost = 0;
};

Packaging::Packaging(int packaging_units, float packaging_cost){
    this->packaging_units = packaging_units;
    this->packaging_cost = packaging_cost;
};

Packaging::Packaging(const Packaging& other) {
    // Copy the values of the other object
    this->packaging_units = other.packaging_units;
    this->packaging_cost = other.packaging_cost;
}

float Packaging::calculate_Total_Cost(){
    return packaging_units*packaging_cost;
};

std::string Packaging::display_product_info(){
    return "Packaing Units: "  +std::to_string(packaging_units)+
           "\nCost per Unit: " +std::to_string(packaging_cost)+
           "\nTotal Packaging Cost: " +std::to_string(calculate_Total_Cost());
}

